back/engines/commercial/analysis/app/lib/analysis/auto_tagging_method/sentiment.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+module Analysis
+  class AutoTaggingMethod::Sentiment < AutoTaggingMethod::Base
+    include NLPCloudHelpers
+
+    TAG_TYPE = 'sentiment'
+    POSITIVE_THRESHOLD = 0.8
+    NEGATIVE_THRESHOLD = 0.95
+    POSITIVE_NAME = 'sentiment +'
+    NEGATIVE_NAME = 'sentiment -'
+
+    protected
+
+    # Use `execute` on the parent class to actually use the method
+    def run
+      positive_tag, negative_tag = sentiment_tags
+
+      total_inputs = analysis.inputs.size
+
+      analysis.inputs.includes(:author).each_with_index do |input, i|
+        update_progress(i / total_inputs.to_f) if i % 5 == 0
+
+        locale = deduct_locale(input)
+        nlp = nlp_cloud_client_for(
+          'distilbert-base-uncased-finetuned-sst-2-english',
+          locale
+        )
+
+        result = nil
+
+        text = input_to_text.execute(input).values.join("\n")
```

Comment:
It could be. For now `InputToText` only converts an individual input. Depending on what you want the model to do, might make sense to format things somewhat differently. That also applies to an individual input, of course. Will leave it like this for now, simple enough.